brave/brave-core
# Add support for dynamic, express and expirable P3A metrics, add NTP SI/creative metrics

Resolves brave/brave-browser#24318
Resolves brave/brave-browser#24319
Resolves brave/brave-browser#24609.

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

0) Start with fresh profile, ensure ads is disabled, ensure no metrics with the prefix `creativeInstanceId` exist in the local state page.
1) Open new tabs until an ad is shown. Verify one creative metric with the suffix `views` exists in the local state page.
2) Click on the creative, verify one creative metric with the suffix `clicks` exists in the local state page.
3) Verify that the metrics are sent (by using a mitm proxy), with the header `X-Brave-P3A-Creative` set to `true`.
4) Close browser, advance the time by one day, open browser, ensure the metrics are resent.
5) Close browser, advance the time by seven days, open browser, ensure metrics are not sent (due to expiry).
